### PR TITLE
Bump versions for 1.5

### PR DESCRIPTION
**Pull request by robertbastian**
https://github.com/unicode-org/icu4x/issues/4925

### CODE REVIEW

File: tutorials/rust/harfbuzz/Cargo.toml
Comment (robertbastian) [inline tutorials/rust/harfbuzz/Cargo.toml:12]: note: I've set `icu_harfbuzz` to 0.6.0 to align with the used `harfbuzz` version

**Review by Manishearth — APPROVED** [review]

File: documents/process/release.md
Comment (sffc) [inline documents/process/release.md:59]: Issue: I assume you deleted this from the guide since we now have our branching approach, but it's still the case that if we need to fix something after a release, we want to make it easy to patch on both main and on the release branch.

File: ffi/capi/src/data_struct.rs
Comment (sffc) [inline ffi/capi/src/data_struct.rs:5]: I'll let @Manishearth approve this feature gate change

File: ffi/harfbuzz/Cargo.toml
Comment (sffc) [inline ffi/harfbuzz/Cargo.toml:8]: Question: why `0.6` when it was previously `0.1`

File: tutorials/rust/baked/build.rs
Comment (sffc) [inline tutorials/rust/baked/build.rs:19]: Issue: `RuntimeManual` maps to `LocaleFamily::with_descendants`, `DeduplicationStrategy::Maximal`, and `RuntimeFallbackLocation::External`

**Review by sffc — CHANGES_REQUESTED** [review]
^

File: ffi/harfbuzz/Cargo.toml
Comment (robertbastian) [reply ffi/harfbuzz/Cargo.toml:8]: to align with the `harfbuzz` version

File: tutorials/rust/baked/build.rs
Comment (robertbastian) [reply tutorials/rust/baked/build.rs:19]: doesn't matter, it's an example

File: documents/process/release.md
Comment (robertbastian) [reply documents/process/release.md:59]: But I don't want to wait another 14 days to make breaking changes :)

File: ffi/capi/src/data_struct.rs
Comment (robertbastian) [reply ffi/capi/src/data_struct.rs:5]: without this there are dead code warnings in the harfbuzz tutorial

File: ffi/harfbuzz/Cargo.toml
Comment (sffc) [reply ffi/harfbuzz/Cargo.toml:8]: I don't see why we should try to sync with the `harfbuzz` version; we could decide to make our own changes that are not synchronized with the `harfbuzz` version. Like, maybe we add support for one of the other harfbuzz crates.

File: ffi/capi/src/data_struct.rs
Comment (sffc) [reply ffi/capi/src/data_struct.rs:5]: If we want to feature-gate the whole module, it should probably be in lib.rs

If we want to feature-gate specific APIs, then feature-gate specific APIs

File: tutorials/rust/baked/build.rs
Comment (sffc) [reply tutorials/rust/baked/build.rs:19]: ok... except if you use `LocaleFamily::single`, that's really a power user feature, so it probably shouldn't be highlighted in the tutorial

File: ffi/capi/src/data_struct.rs
Comment (robertbastian) [reply ffi/capi/src/data_struct.rs:5]: It should not be on the module, because from the module's view, the `data_struct` module has nothing to do with decimal. All *current* methods on `ICU4XDataStruct` are behind that feature, so, I'd need to sprinkle the feature gate all over this file. Having it at the top is easier.

File: ffi/harfbuzz/Cargo.toml
Comment (robertbastian) [reply ffi/harfbuzz/Cargo.toml:8]: At the moment, `icu_harfbuzz` is an implementation of `harfbuzz-traits` 0.6.0.

File: tutorials/rust/baked/build.rs
Comment (robertbastian) [reply tutorials/rust/baked/build.rs:19]: Can we resolve tutorial nits later and unblock the release?

File: ffi/capi/src/data_struct.rs
Comment (Manishearth) [reply ffi/capi/src/data_struct.rs:5]: Fine with this for now

File: tutorials/rust/baked/build.rs
Comment (sffc) [reply tutorials/rust/baked/build.rs:19]: Ok

**Review by sffc — APPROVED** [review]

**Pull request merged**
